### PR TITLE
Add support beanie backend for Mongo

### PR DESCRIPTION
**Pull request by shepilov-vladislav**

### CODE REVIEW

**Review by ADR-007 — APPROVED** [review]
Thank you a lot for your PR! It helped me a lot!

I tried to use it in my project, and found some issues, and want to share my experience just in case it will be useful. Please feel free to leave my suggestions out of the scope of the current PR.

**Comment by ADR-007** [comment]
I can provide PR with my changes on top of this one, if it has any chance of being merged...

File: tests/conftest.py [hunk 1]
```diff
@@ -3,13 +3,13 @@
...
 import pytest
 import pytest_asyncio
-from httpx import AsyncClient
+from httpx import ASGITransport, AsyncClient
 from pydantic import field_validator
 
 
 @pytest_asyncio.fixture(scope="function")
 async def test_client(app):
-    async with AsyncClient(app=app, base_url="http://test") as async_test_client:
```
Comment (arthurio) [inline tests/conftest.py:12]: I'm not famliar with `ASGITransport`, why is this change required?

**Review by arthurio — COMMENTED** [review]
@shepilov-vladislav Sorry for my poor response time, see also https://github.com/arthurio/fastapi-filter/discussions/602

That being said, this PR looks pretty good and I'd be happy to merge it. @ADR-007 can you add your changes first?

File: fastapi_filter/contrib/beanie/filter.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        ):
+            if not value:
+                # Empty string should return [] not ['']
+                return []
+            return list(value.split(","))
+        return value
+
+    def _get_filter_conditions(self, nesting_depth: int = 1) -> list[tuple[Mapping[str, Any], Mapping[str, Any]]]:
+        filter_conditions: list[tuple[Mapping[str, Any], Mapping[str, Any]]] = []
+        for field_name, value in self.filtering_fields:
```
[the remark this answers is not in the document]
Comment (shepilov-vladislav) [reply fastapi_filter/contrib/beanie/filter.py:85]: Added https://github.com/arthurio/fastapi-filter/pull/591/files#diff-148fb7a4eaa0bcf2b90ed5de784293628df9d54e4cd4ce5f103503b8dfd61aadR50-R56

**Comment by shepilov-vladislav** [comment]
@ADR-007 and @arthurio Can you review again?

File: fastapi_filter/contrib/beanie/filter.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+                            (
+                                {f"{field_name}.{sub_field_name}": sub_value},
+                                {"fetch_links": True, "nesting_depth": nesting_depth, **part_options},
+                            )
+                        )
+
+            elif "__" in field_name:
+                stripped_field_name, operator = field_name.split("__")
+                search_criteria = _odm_operator_transformer[operator](value)
+                filter_conditions.append(({stripped_field_name: search_criteria}, {}))
```
Comment (ADR-007) [inline fastapi_filter/contrib/beanie/filter.py:109]: I would like to have `getattr(self.Constants.model, stripped_field_name)` here to make it work with custom attributes here as well

**Review by ADR-007 — APPROVED** [review]

File: fastapi_filter/contrib/beanie/filter.py (hunk 3 above)
Comment (shepilov-vladislav) [reply fastapi_filter/contrib/beanie/filter.py]: @ADR-007 did it

Comment (ADR-007) [reply fastapi_filter/contrib/beanie/filter.py:102]: Thank you!

**Review by arthurio — APPROVED** [review]
This looks awesome, thanks a lot for the contribution!
